streambed-logged/src/lib.rs

```diff
@@ -120,19 +121,21 @@ impl FileLog {
         read_buffer_size: usize,
         compaction_write_buffer_size: usize,
         write_buffer_size: usize,
+        max_record_size: usize,
```

Comment:
We now take a max record size which defaults to 8K (the write buffer size). This is so that we can avoid infinitely allocating memory when decoding. :-)